include/oneapi/dpl/internal/random_impl/discard_block_engine.h

```diff
@@ -166,22 +173,7 @@ class discard_block_engine
 
     template <class CharT, class Traits, class __Engine, std::size_t __P, std::size_t __R>
     friend ::std::basic_istream<CharT, Traits>&
-    operator>>(::std::basic_istream<CharT, Traits>& __is, discard_block_engine<__Engine, __P, __R>& __e)
-    {
-        internal::save_stream_flags<CharT, Traits> __flags(__is);
-
-        __is.setf(std::ios_base::dec);
-
-        __Engine __engine_;
-        std::size_t __n_;
-        if (__is >> __engine_ >> __n_)
-        {
-            __e.engine_ = __engine_;
-            __e.n_ = __n_;
-        }
-
-        return __is;
-    }
+    operator>>(::std::basic_istream<CharT, Traits>&, discard_block_engine<__Engine, __P, __R>&);
```

Comment:
Should I also move definitions of output operators and !=, == outside of the discard_block_engine and subtract_with_carry_engine classes? And should I do that for linear_congruential_engine as well?

Reply:
yes, please, align operators>> and << definitions. (== and != should be ok to leave as is)